elastic/kibana
# [Index pattern field editor] Painless syntax error handling

This PR adds Painless semantic syntax error to the index pattern field editor.

With the previous work from @alisonelizabeth to surface syntax errors in the UI without making an ES call (https://github.com/elastic/kibana/pull/90233), we now have a much better UX to prevent an invalid runtime field to be saved.

### How it works

When the user clicks the "Save" button we make a call to ES with an empty query

```
GET /index/_search
{
    "runtime_mappings": {
        "field": {
            "type": "long",
            "script": "emit(doc['foo']"
        }
    },
    "size": 0,
    "query": {
        "match_none": {}
    }
}
```

We then _only_ look if there are errors on any of the failing shards that are `script_exception`. Once we get one error we exit early and return that error to the UI. 

### How to test

* Write a script with an unknown method

```
emit('hello').someUnknownMethod()
```

* Click "Save"
* The field should not save and there should be error detail below the script field

<img width="824" alt="Screenshot 2021-02-09 at 19 21 12" src="https://user-images.githubusercontent.com/2854616/107416549-83cd9500-6b0c-11eb-9a38-4e76e72837aa.png">
